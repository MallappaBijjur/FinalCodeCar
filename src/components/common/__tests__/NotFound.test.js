import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup } from '@testing-library/react';
import rerender from 'react-test-renderer';
import NotFound from '../NotFound404';

describe("NotFound", () => {

	const originalError = console.error
	beforeAll(() => {
		console.error = (...args) => {
			if (/Warning.*not wrapped in act/.test(args[0])) {
				return
			}
			originalError.call(console, ...args)
		}
	})

	afterAll(() => {
		console.error = originalError
	});

	afterEach(cleanup);
	
	it("renders without crashing", ()=>{
		const div = document.createElement("div");
		ReactDOM.render(<NotFound />, div);
	});
	
	it('renders Filter component', () => {
        const { getByText } = render(<NotFound />);   
		const linkElement = getByText(/404 - Not Found/i);
		expect(linkElement).toBeInTheDocument();
	});
	
	it("matches snapshot", ()=> {
		const tree = rerender.create(<NotFound />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});




