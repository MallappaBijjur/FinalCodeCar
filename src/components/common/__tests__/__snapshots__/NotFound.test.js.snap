// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound matches snapshot 1`] = `
Array [
  <header
    className="header"
  >
    <div>
      <img
        alt="Auto1LOgo"
        className="header__logo"
        src="auto1logo.jpg"
      />
    </div>
    <div
      className="header__nav"
      data-testid="header"
    >
      <a
        className="header__navlink"
        href="#purchase"
      >
        Purchase
      </a>
      <a
        className="header__navlink"
        href="#myOrders"
      >
        My Orders
      </a>
      <a
        className="header__navlink"
        href="#Sell"
      >
        Sell
      </a>
    </div>
  </header>,
  <div
    className="error__container"
  >
    <div
      className="error"
    >
      <img
        alt="Auto1"
        className="logo"
        src="auto1logo.jpg"
      />
      <h2
        className="error__text--dark error__text-large"
      >
        404 - Not Found
      </h2>
      <p
        className="error__text--medium error__text--dark"
      >
        Sorry, the page you are looking for does not exist.
      </p>
      <p
        className="error__text--medium"
      >
        You can always go back to the
        <a
          className="error__link"
          href="/"
        >
           homepage
        </a>
      </p>
    </div>
  </div>,
  <footer
    className="footer"
  >
    <p
      className="footer__text"
    >
      Â© AUTO1 Group 2018
    </p>
  </footer>,
]
`;
